Screens/NewRace.js

```diff
@@ -0,0 +1,170 @@
+import React, { Component } from 'react'
+import {
+  Alert,
+  Button,
+  Picker,
+  StyleSheet,
+  Text,
+  TextInput,
+  View
+} from 'react-native'
+
+class NewRace extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      codex: '',
+      gender: 'Female',
+      temperature: '',
+      precipitation: 'None',
+      course: '',
+      racers: 6,
+      position: 'Start Gate',
+    }
+  }
+
+  static navigationOptions = {
+    title: 'New Race'
+  }
+
+  _handleBtnPress() {
+    const { navigate } = this.props.navigation
+    if (this.state.codex       === '' ||
+        this.state.temperature === '' ||
+        this.state.course      === ''
+       ) {
+      Alert.alert('Error: please enter a value for all text fields.')
+    }
+    else {
+      navigate('LineupForm', { title: this.state.position, props: this.state })
+    }
+  }
+
+  render() {
+    return (
+      <View style={styles.container}>
+        <View style={styles.row}>
+          <Text style={styles.text}>
+            Race Codex:
+          </Text>
+          <TextInput
+            style={styles.input}
+            value={this.state.codex}
+            placeholder={'N/a'}
+            onChangeText={(text) => this.setState({codex: text})}
```

Comment:
>onChangeText [](start = 12, length = 12)

You could experiment with the `onEndEditing` event here - it's fired only when the TextInput loses focus (so not as much activity over the native/JS bridge).